### ISSUE TITLE
Bug classement

### ISSUE THREAD

**Issue by MatthieuDEVALLE** [matthieu637/cpp-2a-info#35]
Bonjour,

Nous avons fait une séance d'entrainement cette après-midi et les classements changent en fonction de la personne (et sur plusieurs parties)

Par exemple:
Sur mon PC:
`[('MatthieuDevalle', 'devalle1u:CPP-HZN-W7-032:100.64.214.78', 5582),
 ('GautierPrevoteaux', 'prevotea1u:CPP-HZN-W7-029:100.64.214.92', 3828), 
('PaulFestor', 'festor4u:CPP-HZN-W7-015:100.64.214.69', 2529),
 ('JosephineMougel', 'Joséphine:sosinfo-PC:46.193.65.53', 2519), 
('QuentinFabiani', 'fabiani4u:CPP-HZN-W7-008:100.64.214.85', 2205),
 ('MathieuLaprie', 'laprie3u:CPP-HZN-W7-021:100.64.214.91', 2157), 
('EliaDeBiasi', 'debiasi3u:CPP-HZN-W7-037:100.64.214.114', 2020),
 ('MarionRoth', 'Adrien:DESKTOP-J46RECT:46.193.65.53', 1897), 
('SimonHerbe', 'porpret:cpp-porpret01:90.26.61.10', 5024),
 ('PierreImhoff', 'pierre:pc-pierre:193.50.135.193', 5015),
 ('DylanAlibayMandjyDitFadjou', 'alibay3u:CPP-HZN-W7-039:100.64.214.118', 3820)]`

Sur celui de  Paul:
`In [1]: r.fin()
Out[1]: 
{'temps': 0,
 'classement': [('MatthieuDevalle',
   'devalle1u:CPP-HZN-W7-032:100.64.214.78',
   5582),('GautierPrevoteaux', 'prevotea1u:CPP-HZN-W7-029:100.64.214.92', 3828),
  ('DylanAlibayMandjyDitFadjou','alibay3u:CPP-HZN-W7-039:100.64.214.118',3820),
  ('PaulFestor', 'festor4u:CPP-HZN-W7-015:100.64.214.69', 2529),
  ('JosephineMougel', 'Joséphine:sosinfo-PC:46.193.65.53', 2519),
  ('MarionRoth', 'Adrien:DESKTOP-J46RECT:46.193.65.53', 2219),
  ('QuentinFabiani', 'fabiani4u:CPP-HZN-W7-008:100.64.214.85', 2205),
  ('MathieuLaprie', 'laprie3u:CPP-HZN-W7-021:100.64.214.91', 2157),
  ('EliaDeBiasi', 'debiasi3u:CPP-HZN-W7-037:100.64.214.114', 2020),
  ('SimonHerbe', 'porpret:cpp-porpret01:90.26.61.10', 5024),
  ('PierreImhoff', 'pierre:pc-pierre:193.50.135.193', 5015)]}`

**Comment by matthieu637** [comment]
Est-ce que tu as bien attendu 1 seconde environ après la fin de partie pour appeller r.fin() ? Il se peut que des transactions soient encore en cours d'annulation.

**Comment by MatthieuDEVALLE** [comment]
Oui, j'ai mis un Time.sleep(3) pour être sûr mais peut être que ce n'est pas le cas de Paul.Pour être complètement certains je vais attendre 5s.

**Comment by matthieu637** [comment]
3 secondes sont suffisantes, de mémoire c'est de l'ordre d'une seconde maximale.
Est-ce que l'un de vous ne modifierait pas l'affichage de la liste?

**Comment by MatthieuDEVALLE** [comment]
Pour ma part, j'ai uniquement:

time.sleep(3)
classement=r.fin()['classement']
print(classement)

Et après je modifie pour pouvoir l'afficher correctement dans tkinter.

**Comment by Paul-543NA** [comment]
Bonjour, nous avons refait des tests cette après midi sur un total de 8 parties (ouvertes à tous les étudiants de la promo) et en passant à un sleep de 5 secondes pour Matthieu et 3 secondes pour moi et nous n'avons eu aucun problème d'incohérence des classements. Le soucis relevé hier venait donc sûrement d'une attente trop courte avant de récupérer le classement.
